app-libs/stf/src/stf_sgx.rs

```diff
@@ -240,6 +247,41 @@ where
 	}
 }
 
+impl<Call, Getter, State, Runtime> EncointerSchedulerPalletInterface<State>
+	for Stf<Call, Getter, State, Runtime>
+where
+	State: SgxExternalitiesTrait,
+	Runtime: pallet_encointer_scheduler::Config + pallet_encointer_ceremonies::Config,
+{
+	fn update_ceremony_phase(state: &mut State, next_ceremony_phase: &[u8]) {
+		state.execute_with(|| {
+			let current_ceremony_phase = current_ceremony_phase();
+			match CeremonyPhaseType::decode(&mut &*next_ceremony_phase) {
+				Ok(decoded_next_ceremony_phase) => {
+					if current_ceremony_phase.is_none()
+						|| decoded_next_ceremony_phase != current_ceremony_phase.unwrap_or_default()
+					{
+						info!(
```

Comment:
No, that I look at this, maybe we need to make sure that the logic is really sound. Under what circumstances can the `current_ceremony_phase` be `None`? Does it make sense to execute `OnCeremonyPhaseChange`, when this has been none before? I am not sure.

Reply:
I was just thinking, if this is none, then I suppose it is not the only state that is not yet in the sidechain. I don't have now the detailed view to judge if I am stating an actual problem. We can also post-pone this problem to somewhere in the future.

Reply:
I think we can just leave the  `is_none` check away. Reasoning: 
If `current_ceremony_phase` is `None` then the statement

```decoded_next_ceremony_phase != current_ceremony_phase.unwrap_or_default()```  equals
```decoded_next_ceremony_phase != None``` . So the ceremony phase is update if `decoded_next_ceremony_phase` is not `None`. Which is fine. If both are `None`, then it doesn't make sense to update anyway.

So the check `is_none` is obsolete IMO.

Reply:
Btw: I really like the unwrap_or_default. I think it's the better choice than a simple `unwrap` 👍

Reply:
My bad, you are correct: unwarp_or_default does not return None but a default value for the Value of Some. That's not good.